cmake/Linters.cmake

```diff
@@ -83,7 +80,49 @@ if(NOT ${CLANG_FORMAT_CMD})
     )
   endif()
 else()
-  message(WARNING "clang-format executable not found. Disabling clang-format checks. clang-format can be installed with \"pip3 install --user --upgrade clang-format\"")
+  message(WARNING "clang-format executable not found. Disabling clang-format checks. clang-format can be installed with \"pip3 install --user --upgrade clang-format\" or your distribution's package manager.")
+endif()
+
+
+#################### CLANG-TIDY
+
+# TODO: Add files here once clang-tidy is fast enough.
+# file(
+#   GLOB_RECURSE P4C_CLANG_TIDY_LINT_LIST  FOLLOW_SYMLINKS
+# )
+# add_clang_tidy_files(${P4C_SOURCE_DIR} "${P4C_CLANG_TIDY_LINT_LIST}")
+
+find_program(CLANG_TIDY_CMD clang-tidy)
+
+if(CLANG_TIDY_CMD)
+  if (NOT CMAKE_EXPORT_COMPILE_COMMANDS)
+    message(WARNING "CMAKE_EXPORT_COMPILE_COMMANDS is not set ON, clang-tidy remains disabled.")
+  endif()
+
+  # Retrieve the global clang-tidy property.
+  get_property(CLANG_TIDY_FILES GLOBAL PROPERTY CLANG_TIDY-files)
+  if(DEFINED CLANG_TIDY_FILES AND CMAKE_EXPORT_COMPILE_COMMANDS)
+
+    # Write the list to a file.
+    # We need to do this as too many files will reach the shell argument limit.
+    set(CLANG_TIDY_TXT_FILE ${P4C_BINARY_DIR}/clang_tidy_files.txt)
+    list(SORT CLANG_TIDY_FILES)
+    file(WRITE ${CLANG_TIDY_TXT_FILE} "${CLANG_TIDY_FILES}")
+    add_custom_target(
+      clang-tidy
+      COMMAND xargs -a ${CLANG_TIDY_TXT_FILE} -r -d '\;' ${CLANG_TIDY_CMD} -p ${CMAKE_BINARY_DIR}
```

Comment:
Maybe we could make advantage of the `-P` option of `xargs` to run multiple invocations in parallel.

Reply:
Doesn't seem to work out of the box. We can try this in subsequent PRs.